container-core/src/main/java/com/yahoo/container/jdisc/state/StateHandler.java

```diff
@@ -212,6 +216,56 @@ private ObjectNode buildJsonForConsumer(String consumer) {
         return ret;
     }
 
+    private byte[] buildPrometheusForConsumer(String consumer) {
+        var snapshot = getSnapshot();
+        if (snapshot == null)
+            return new byte[0];
+
+        var timestamp = snapshot.getToTime(TimeUnit.MILLISECONDS);
+        var builder = new StringBuilder();
+        builder.append("# NOTE: THIS API IS NOT INTENDED FOR PUBLIC USE\n");
+        for (var tuple : collapseMetrics(snapshot, consumer)) {
```

Comment:
If my understanding of `collapseMetrics` _and_ the [Prometheus exposition docs](https://prometheus.io/docs/instrumenting/exposition_formats/#grouping-and-sorting) is correct, I think we need to add an extra aggregation step that groups all samples for the same metric _name_ under the same `HELP`/`TYPE` umbrella:

> All lines for a given metric must be provided as one single group, with the optional `HELP` and `TYPE` lines first (in no particular order). Beyond that, reproducible sorting in repeated expositions is preferred but not required, i.e. do not sort if the computational cost is prohibitive.
>
> Each line must have a unique combination of a metric name and labels. Otherwise, the ingestion behavior is undefined.

I.e. if we have two distinct samples (different dimensions) for a gauge `foo`, the different time series this metric results in need to be grouped together in the output:
```
...
# TYPE foo_sum untyped
foo_sum{bar="baz"} 123
foo_sum{bar="zoid"} 456
# TYPE foo_count untyped
foo_count{bar="baz"} 2
foo_count{bar="zoid"} 4
...
```